### PR TITLE
Feature/redux - basic structure + needed libraries

### PR DESCRIPTION
**Pull request by KitsuneAkvma**
- Created userSlice to keep current user data.
For now it is only name and email
- Created utility states,reducers and selectors for
state of `LogoutModal` and `isLoading` which indicates
if page is loading or not.
- Changed name `modalsSlice` to `utilitySlice` to keep it more readable
and universal for the sake of those states purpose.

### CODE REVIEW

File: src/main.jsx [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+import { ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
 
 ReactDOM.createRoot(document.getElementById('root')).render(
   <>
-    <Provider store={Store}>
-      <App />
-    </Provider>
+    {' '}
+    <PersistGate loading="null" persistor={persistor}>
```
Comment (MalgorzataMarczynska) [inline src/main.jsx:14]: persist wewnątrz providera?

File: src/redux/Slices/finance/financeSlice.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const handleRejected = (state, action) => {
+  state.isLoading = false;
+  state.data = [];
+  state.error = action.payload;
+};
+
+const financeSlice = createSlice({
+  name: 'finance',
+  initialState,
+  reducers: {
```
Comment (MalgorzataMarczynska) [inline src/redux/Slices/finance/financeSlice.js:22]: jesteś pewny, ze można w jednym slicie łaczyć wersję deprecated reducers z extrareducerami(im chyba buildera brakuje)?

File: src/redux/Slices/finance/operations.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+
+const getFinanceData = createAsyncThunk((userID, thunkAPI) => {
```
Comment (MalgorzataMarczynska) [inline src/redux/Slices/finance/operations.js:5]: tu nie można od razu exporta dać zamiast tego na dole?

Comment (KitsuneAkvma) [reply src/redux/Slices/finance/operations.js:5]: Można ale lepiej utrzymać spójność kodu.

**Review by MalgorzataMarczynska — APPROVED** [review]

**Pull request merged**